### PR TITLE
commits for 64 bit builds

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/drawing/scrolling_text.c [hunk 1]
```diff
@@ -130,6 +130,194 @@ static void scrolling_text_format(utf8 *dst, rct_draw_scroll_text *scrollText)
 
 extern bool TempForScrollText;
 
+sint16 _scrollpos0[] = {
```
Comment (IntelOrca) [inline src/drawing/scrolling_text.c]: const?

**Comment by janisozaur** [comment]
updated

**Comment by marijnvdwerf** [comment]
For reference: this are the different scrollpos-types:
rct_scrollpos

**Comment by marijnvdwerf** [comment]
I think having the scrollposses defined like this would make it more obvious that the elements are coordinates.
```
static const rct_xy16 _scrollpos37[] = {
    { 32, 13},
    { 33, 12},
    { 34, 12},
    { 35, 12},
    { 36, 12},
    { 37, 11},
    { 38, 11},
    { 39, 11},
    { 40, 10},
    { 41, 10},
    { 42, 10},
    { 43,  9},
    { 44,  9},
    { 45,  8},
    { 46,  8},
    { 47,  7},
    { 48,  7},
    { 49,  6},
    { 50,  6},
    { 51,  5},
    { 52,  5},
    { 53,  4},
    { 54,  3},
    { 55,  3},
    { 56,  2},
    { 57,  1},
    { 58,  0},
    { -1, -1},
};
```

File: src/peep/peep.c [hunk 2]
```diff
@@ -350,6 +350,167 @@ static const ride_rating NauseaMaximumThresholds[] = {
...
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	_97e1bc_21,
```
Comment (marijnvdwerf) [inline src/peep/peep.c]: `// RIDE_TYPE_SPIRAL_SLIDE`

**Comment by janisozaur** [comment]
I'm confused, how big are elements in each `_scrollpos`? When I cast them to 8 bits, I get:
```
 {96, 3},
 {33, 3},
 {34, 3},
 {35, 3},
 {36, 3},
 {229, 2},
 {230, 2},
 {231, 2},
 {168, 2},
 {169, 2},
 {170, 2},
 {107, 2},
 {108, 2},
 {45, 2},
 {46, 2},
 {239, 1},
 {240, 1},
 {177, 1},
 {178, 1},
 {115, 1},
 {116, 1},
 {53, 1},
 {246, 0},
 {247, 0},
 {184, 0},
 {121, 0},
 {58, 0},
```
for `_scrollpos37`

**Comment by marijnvdwerf** [comment]
Based on: https://github.com/janisozaur/OpenRCT2/blob/89c365c301e7f9560d9ce850206d5740839319e7/src/drawing/scrolling_text.c#L481-L487: 

`element % 64` is the x position, `element / 64` is y.

Quick output from the script I used to create the images:
http://pastebin.com/kNGqSTet

**Comment by janisozaur** [comment]
But this breaks when I want to use `rct_xy8.xy` as an offset, when it is getting actually used in https://github.com/OpenRCT2/OpenRCT2/blob/a6d0e69/src/drawing/scrolling_text.c#L223 and would introduce runtime overhead, if I wanted to shift the bits in place.

**Comment by janisozaur** [comment]
@marijnvdwerf check https://github.com/janisozaur/OpenRCT2/commit/5ec65487542854a0edb4d1c43ec8451beeb71c4e

**Comment by IntelOrca** [comment]
_scrollpos is an array of uint16 offsets into bitmap. So yes, 0 to 63 is the first scan line, 64 to 127 is the second scan line. You can't separate them into x, y bytes without changing the way they are used.

**Comment by janisozaur** [comment]
How about now? The way @marijnvdwerf suggested.

File: src/drawing/scrolling_text.c [hunk 3]
```diff
@@ -475,10 +1547,10 @@ void scrolling_text_set_bitmap_for_ttf(utf8 *text, int scroll, uint8 *bitmap, co
...
 		// Skip any none displayed columns
 		if (scroll == 0) {
-			sint16 scrollPosition = *scrollPositionOffsets;
-			if (scrollPosition == -1) return;
-			if (scrollPosition > -1) {
-				uint8 *dst = &bitmap[scrollPosition];
+			rct_xy8 scrollPosition = *scrollPositionOffsets;
+			if (scrollPosition.xy == (uint16)-1) return;
+			if ((sint16)scrollPosition.xy > -1) {
+				uint8 *dst = &bitmap[scrollPosition.x + scrollPosition.y * 64];
```
Comment (janisozaur) [inline src/drawing/scrolling_text.c]: While the variables are stored nicely now, the drawback is multiplication which has to be done here. But perhaps compiler can figure it out and optimise or use the `lea` for calculating the address here.

File: src/drawing/scrolling_text.c [hunk 4]
```diff
@@ -130,6 +130,1266 @@ static void scrolling_text_format(utf8 *dst, rct_draw_scroll_text *scrollText)
 
 extern bool TempForScrollText;
 
+static const rct_xy8 _scrollpos0[] = {
+	{ 35, 12},
```
Comment (IntelOrca) [inline src/drawing/scrolling_text.c]: Can you put a space before the end brace, kind of irritating me :smile:

**Comment by janisozaur** [comment]
I have used `SCROLL_POS` macro you suggested, just in case. I hope @marijnvdwerf is OK with that.

**Comment by marijnvdwerf** [comment]
As long as it's readable, it's fine with me. Pretty much anything is better than the raw pixel offset :)

File: src/ride/ride.c [hunk 5]
```diff
@@ -124,6 +124,438 @@ uint8 gTypeToRideEntryIndexMap[TYPE_TO_RIDE_ENTRY_SLOTS];
 
 #pragma endregion
 
+const rct_ride_properties gRideProperties[91] = {
+		{RIDE_ENTRY_FLAG_0 | RIDE_ENTRY_FLAG_NO_INVERSIONS | RIDE_ENTRY_FLAG_NO_BANKED_TRACK |
```
Comment (marijnvdwerf) [inline src/ride/ride.c]: Shouldn't these be `RIDE_TYPE` flags?

File: src/ride/ride.c [hunk 6]
```diff
@@ -124,6 +124,438 @@ uint8 gTypeToRideEntryIndexMap[TYPE_TO_RIDE_ENTRY_SLOTS];
 
 #pragma endregion
 
+const rct_ride_properties gRideProperties[91] = {
```
Comment (marijnvdwerf) [inline src/ride/ride.c]: Missing `rct: 0x0097CF40` comment

File: src/ride/ride.h [hunk 7]
```diff
@@ -883,6 +883,16 @@ enum {
 	RIDE_SETTING_RIDE_TYPE,
 };
 
+typedef struct rct_ride_properties {
+	uint32 flags;
+	uint8 minValue;
+	uint8 maxValue;
+	uint8 maxBrakesSpeed;
+	uint8 accelaration;
```
Comment (marijnvdwerf) [inline src/ride/ride.h]: acceleration

File: src/ride/ride.h [hunk 8]
```diff
@@ -883,6 +883,16 @@ enum {
 	RIDE_SETTING_RIDE_TYPE,
 };
 
+typedef struct rct_ride_properties {
+	uint32 flags;
+	uint8 minValue;
```
Comment (marijnvdwerf) [inline src/ride/ride.h]: Struct members usually are lowercase + underscored, right?

File: src/ride/ride.c (hunk 5 above)
Comment (janisozaur) [inline src/ride/ride.c]: Hmm, you may be right.

In that case, I was misled by this comment:
```
// Constants used by the ride_type->flags property at 0x008
```
https://github.com/janisozaur/OpenRCT2/blob/integrate2/src/ride/ride.h#L413

**Comment by marijnvdwerf** [comment]
RideProperties should probably be in `ride_data.c`, and it would be nice if they had the same comments prefixed as `RideTypePossibleTrackConfigurations`. Makes it a lot easier to read.

File: src/ride/ride_data.c [hunk 9]
```diff
@@ -1729,3 +1729,508 @@ const rating_tuple RideRatings[91] = {
 	{	50,		30,	30	},	// RIDE_TYPE_59
 	{	50,		30,	10	},	// RIDE_TYPE_LIM_LAUNCHED_ROLLER_COASTER
 };
+
+// rct2: 0x0097CF40
+const rct_ride_properties gRideProperties[91] = {
```
Comment (marijnvdwerf) [inline src/ride/ride_data.c]: Why are some constants prefixed with `g`? I'd expect it to be for global *variables* only.

**Comment by janisozaur** [comment]
I've amended last commit to rename it to `RideProperties`

File: src/ride/ride_data.c [hunk 10]
```diff
@@ -1729,3 +1729,508 @@ const rating_tuple RideRatings[91] = {
...
+		 RIDE_TYPE_FLAG_SUPPORTS_MULTIPLE_TRACK_COLOUR,
+				0,  0,   0,  0},  // RIDE_TYPE_59
+		{RIDE_TYPE_FLAG_HAS_TRACK_COLOUR_MAIN | RIDE_TYPE_FLAG_HAS_TRACK_COLOUR_ADDITIONAL |
+		 RIDE_TYPE_FLAG_HAS_TRACK_COLOUR_SUPPORTS | RIDE_TYPE_FLAG_HAS_LEAVE_WHEN_ANOTHER_VEHICLE_ARRIVES_AT_STATION |
+		 RIDE_TYPE_FLAG_CAN_SYNCHRONISE_ADJACENT_STATIONS | RIDE_TYPE_FLAG_HAS_G_FORCES |
+		 RIDE_TYPE_FLAG_HAS_DATA_LOGGING | RIDE_TYPE_FLAG_HAS_DROPS | RIDE_TYPE_FLAG_HAS_LOAD_OPTIONS |
+		 RIDE_TYPE_FLAG_20 | RIDE_TYPE_FLAG_26 | RIDE_TYPE_FLAG_27 | RIDE_TYPE_FLAG_HAS_TRACK | RIDE_TYPE_FLAG_30 |
+		 RIDE_TYPE_FLAG_SUPPORTS_MULTIPLE_TRACK_COLOUR,
+				10, 31,  26, 18},  // RIDE_TYPE_LIM_LAUNCHED_ROLLER_COASTER
+};
```
Comment (marijnvdwerf) [inline src/ride/ride_data.c]: Missing newline

**Pull request merged**